src/NSFW.cpp

```diff
@@ -74,14 +75,21 @@ NSFW::NSFW(const Napi::CallbackInfo &info):
 }
 
 NSFW::~NSFW() {
-  mErrorCallback.Release();
-  mEventCallback.Release();
-
   if (gcEnabled) {
     instanceCount--;
   }
 }
 
+void NSFW::Finalize(Napi::Env env) {
+  if (mRunning) {
+    mFinalizing = true;
+    mRunning = false;
+    Unref();
```

Comment:
I'm assuming we need this here because `StopWorker::OnOk` never triggers if we finalize? Is there a chance we `Finalize` after `StopWorker::OnOk` and `Unref()` twice?

Reply:
Finalize is called just before delete the NFSW object so we need to stop the pollForEvents thread if the watcher is started.
If the watcher is stopped then mRunning is false and we do not need to stop the thread manually, so it is not possible to call `Unref()` twice